src/main/kotlin/pl/edu/agh/server/application/user/UserController.kt

```diff
@@ -17,10 +19,42 @@ class UserController(
     private val userService: UserService,
     private val modelMapper: ModelMapper,
     private val jwtService: JwtService,
+    private val organizationService: OrganizationService,
 ) : BaseControllerUtilities<User>(jwtService) {
+
     @GetMapping
     fun getUser(request: HttpServletRequest): UserDTO {
         val user = userService.getUserByEmail(getUserName(request))
         return modelMapper.map(user, UserDTO::class.java)
     }
+
+    @GetMapping("/organization-roles/{organizationId}")
+    fun getLoggedUserOrganizationRoles(
+        request: HttpServletRequest,
+        @PathVariable organizationId: Long,
+        @RequestParam("email", required = false) email: String?,
+    ): ResponseEntity<List<OrganizationRole>> {
+        val organizationUserRoles: List<OrganizationRole> = when {
+            !email.isNullOrBlank() -> {
+                userService.getOrganizationRoles(organizationId, email)
+            }
+
+            else -> {
+                userService.getOrganizationRoles(organizationId, userService.getUserId(request))
+            }
+        }
+        return ResponseEntity.ok(organizationUserRoles)
+    }
```

Comment:
Troche mi sie nie podoba ze dowolny uzytkownik moze wyciagac dane o dowolnych innych uzytkownikach zmieniajac email poprostu.

Do czego jest uzywany ten endpoint? Jak tylko sprawdzamy tu swoje dostepy to powinno wystarczyc zeby to wyciagnac z requesta, a jak nie, to moze da sie to jakos ograniczyc chociaz?

Reply:
będzie potrzebne dla widoku użytkowników dla admina/heada ale rozdizele to faktycznie